tex/gregoriotex-syllable.tex

```diff
@@ -556,7 +556,7 @@
   % took me years to find it!
   \gre@debugmsg{bolshift}{Current glyph: \the\gre@attr@glyph@id}%
   \ifgre@bolshiftsenabled%
-    \gre@calculate@bolshift{\gre@dimen@begindifference}%
+    \gre@calculate@bolshift{\gre@dimen@begindifference}{\gre@firstisalteration}% TODO: compute firstisalteration
```

Comment:
You've still got to know that the first piece is an alteration, which involves doing something in the same basic place as where I calculated `bolextra`.  Frankly, I don't see the advantage of setting a flag vs. calculating a distance.  Both have to be passed to `\gre@calculate@bolshift` either implicitly (as I did) or explicitly (as you did).